### PR TITLE
Update HyperLogLog.cs

### PR DESCRIPTION
**Pull request by yatskovsky**

### CODE REVIEW

File: src/HyperLogLog.net/HyperLogLog.net/HyperLogLog.cs [hunk 1]
```diff
@@ -80,11 +92,14 @@ public int GetCount()
...
                     dv = (int)(M * Math.Log((double)M / v));
             }
-            else if (dvEstimate <= (Pow2_32 * 1 / 30))
+            else if (dvEstimate <= (Pow2_32 * 1 / 30))  // 143,165,576
                 dv = (int)dvEstimate;
             else
             {
-                dv = (int)(Math.Pow(-2, 32) * Math.Log(1 - dvEstimate / Pow2_32));
+                if (dvEstimate <= 1E9)
+                    dv = (int)(-Pow2_32 * Math.Log(1 - dvEstimate / Pow2_32));
```
Comment (saguiitay) [inline src/HyperLogLog.net/HyperLogLog.net/HyperLogLog.cs:100]: My math isn't very good, but according to Bing, `Math.Pow(-2, 32)` equals `Pow2_32`, and not `-Pow2_32`: http://www.bing.com/search?q=(-2)%5E32

File: src/HyperLogLog.net/HyperLogLog.net/HyperLogLog.cs [hunk 2]
```diff
@@ -55,21 +55,33 @@ public HyperLogLog(HashAlgorithm hashAlgorithm, IBytesConverter bytesConverter,
...
+        public void LogHash(ulong hash)
+        {
             var registerIndex = GetRegisterIndex(hash); // binary address of the rightmost b bits
             var runLength = RunOfZeros(hash); // length of the run of zeroes starting at bit b+1
             _registers[registerIndex] = Math.Max(_registers[registerIndex], runLength);
         }
-
+        
+        
+        public double FastPow2(int x)
```
Comment (saguiitay) [inline src/HyperLogLog.net/HyperLogLog.net/HyperLogLog.cs:74]: This method should be private, and should be after the other public methods

Comment (yatskovsky) [inline src/HyperLogLog.net/HyperLogLog.net/HyperLogLog.cs:74]: You're right.

File: src/HyperLogLog.net/HyperLogLog.net/HyperLogLog.cs (hunk 1 above)
Comment (yatskovsky) [inline src/HyperLogLog.net/HyperLogLog.net/HyperLogLog.cs:100]: the formula in the article is -2^32, not (-2)^32. if see it the latter way, it gives negative cardinalities becase log of <1 is always negative

**Pull request merged**